tutorials/introductory/animation_tutorial.py

```diff
@@ -0,0 +1,249 @@
+    del ydata[:]
+    line.set_data(xdata, ydata)
+    return line,
+
+
+def update(frame):
+    # .set_data resets all the data for the line, so we add the new point to
+    # the existing line data and set that again.
+    xdata.append(frame / 30)
+    ydata.append(np.sin(2 * np.pi * frame / 30))
+
+    line.set_data(xdata, ydata)
+    return (line,)
+
+
+ani = animation.FuncAnimation(fig=fig, func=update,
+                              interval=30, init_func=init)
+plt.show()
+
+###############################################################################
+# Animating :class:`~matplotlib.collections.PathCollection`
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# `.pyplot.scatter` returns a :class:`~matplotlib.collections.PathCollection`
+# that can similarly be modified by using the
+# `.collections.PathCollection.set_offsets` function.
+
+fig, ax = plt.subplots()
+rng = np.random.default_rng()
+t = np.linspace(-4, 4, 400)
+a, b = 3, 2
+delta = np.pi / 2
+
+scat = ax.scatter(np.sin(a * t[0] + delta), np.sin(b * t[0]), c="b", s=2)
+ax.grid()
+ax.set_xlim(-1, 1)
+ax.set_ylim(-1, 1)
+
+
+def update(frame):
+    # .set_offsets also resets the entire data for the collection.
+    # Therefore, we create the entire data in each frame to draw
+    x = np.sin(a * t[:frame] + delta)
+    y = np.sin(b * t[:frame])
+    data = np.stack([x, y]).T
+    scat.set_offsets(data)
+    return (scat,)
+
+
+ani = animation.FuncAnimation(fig=fig, func=update, frames=400, interval=30)
+plt.show()
+
+
+###############################################################################
+# Animating :class:`~matplotlib.image.AxesImage`
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# When we plot an image using `.pyplot.imshow`, it returns an
+# :class:`~matplotlib.image.AxesImage` object. The data in this object can also
+# similarly be modified by using the `.image.AxesImage.set_data` method.
+
+fig, ax = plt.subplots()
+rng = np.random.default_rng()
+
+aximg = ax.imshow(rng.uniform(low=0, high=1, size=(10, 10)), cmap="Blues")
+
+
+def update(frame):
+    data = rng.uniform(low=0, high=1, size=(10, 10))
+    aximg.set_data(data)
+    return (aximg,)
+
+
+ani = animation.FuncAnimation(fig=fig, func=update, frames=None, interval=200)
+plt.show()
+
+###############################################################################
+# :class:`~matplotlib.animation.ArtistAnimation`
+# ----------------------------------------------
+#
+# On the other hand, :class:`~matplotlib.animation.ArtistAnimation` can be used
+# to generate animations if we have data on various different artists. This
+# list of artists is then converted frame by frame into an animation.
+
+
+fig, ax = plt.subplots()
+ax.grid()
+rng = np.random.default_rng()
+
+x_frames = rng.uniform(low=0, high=1, size=(100, 120))
+y_frames = rng.uniform(low=0, high=1, size=(100, 120))
+artists = [
+    [ax.scatter(x_frames[:, i], y_frames[:, i], c="b")]
+    for i in range(x_frames.shape[-1])
+]
```

Comment:
The list needs to be nested because `ArtistAnimation` takes in an iterable where each element is a  _collection_ or `Artist` objects and hence need to be iterable. (Ref: https://matplotlib.org/devdocs/api/_as_gen/matplotlib.animation.ArtistAnimation.html#matplotlib.animation.ArtistAnimation) 

For reference, It gives me a `TypeError: 'PathCollection' object is not iterable` error if it isn't nested.